### PR TITLE
Add type-forwarders for Xamarin.Android compatibility to System.Drawing.Common.dll

### LINKED ISSUE #82829: Workaround broken Mono System.Drawing facades
Legacy Mono had [incorrect facades for some System.Drawing types](https://github.com/mono/mono/issues/21239). This was an effect of System.Drawing being the implementation assembly and System.Drawing.Primitives/Common being the type forwarders. Due to the Mono SDK bugs many NuGets for MonoAndroid/Xamarin.iOS incorrectly expects the following types to be present in System.Drawing.Common:

- System.Drawing.Color
- System.Drawing.KnownColor
- System.Drawing.Point
- System.Drawing.PointF
- System.Drawing.Rectangle
- System.Drawing.RectangleF
- System.Drawing.Size
- System.Drawing.SizeF

PR #82618 solved a similar problem for mscorlib. We should consider adding the type forwarders from System.Drawing.Common to System.Drawing.Primitives to offer compatibility for MonoAndroid assemblies.

### PR DESCRIPTION
**Pull request by akoeplinger**
The legacy Xamarin.Android version of System.Drawing.Common.dll contained these types, add forwarders so we don't get a TypeLoadException when using an assembly compiled against that in modern .NET Android.

Fixes #82829

### CODE REVIEW

File: src/libraries/System.Drawing.Common/src/System.Drawing.Common.Forwards.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// This is required for back-compatibility with legacy Xamarin which had these types in System.Drawing.Common.dll
```
Comment (ViktorHofer) [inline src/libraries/System.Drawing.Common/src/System.Drawing.Common.Forwards.cs:4]: You are adding these to the implementation assembly only. That's intentional, right?

Comment (filipnavara) [reply src/libraries/System.Drawing.Common/src/System.Drawing.Common.Forwards.cs:4]: I think that is intentional. We only need the type-forwarders for binary compatibility at runtime. I am not sure how it works with the ApiCompat checks though.

`SystemColors` and `ColorTranslator` have their respective `TypeForwardedTo` in the ref sources as well. These two types were moved between assemblies between .NET Core 2 and 3 IIRC. Do I understand correctly that the implementation assembly previously got the `TypeForwardedTo` for these two types thanks to `IsPartialFacadeAssembly = true`?

Comment (akoeplinger) [reply src/libraries/System.Drawing.Common/src/System.Drawing.Common.Forwards.cs:4]: Actually, we'd still need the IsPartialFacadeAssembly for the other TFMs so we'd need a conditional. I'll just leave it as is.

Comment (ViktorHofer) [reply src/libraries/System.Drawing.Common/src/System.Drawing.Common.Forwards.cs:4]: I'm fine with that approach and it follows what we do in our shims builds (manual type forwards only exposed in src but not ref). Just to get a second pair of eyes, let me cc'ing @ericstj

File: src/libraries/System.Drawing.Common/src/CompatibilitySuppressions.xml [hunk 2]
```diff
@@ -30,6 +30,150 @@
     <Left>lib/netstandard2.0/System.Drawing.Common.dll</Left>
     <Right>lib/net462/System.Drawing.Common.dll</Right>
   </Suppression>
+  <Suppression>
```
Comment (ViktorHofer) [inline src/libraries/System.Drawing.Common/src/CompatibilitySuppressions.xml:33]: Can you please add a comment above this element to indicate that these type forwards exist for back-compatibility with xamarin frameworks?

**Review by ViktorHofer — APPROVED** [review]
LGTM aside from adding a comment to the suppression file.

**Pull request merged**